lightning/src/ln/outbound_payment.rs

```diff
@@ -2238,7 +2244,7 @@ impl_writeable_tlv_based_enum_upgradable!(PendingOutboundPayment,
 	(5, AwaitingInvoice) => {
 		(0, expiration, required),
 		(2, retry_strategy, required),
-		(4, max_total_routing_fee_msat, option),
```

Comment:
Is this breaking backward compatibility?

Matt taught me how this works a few weeks ago, but I’m not completely sure if I’m understanding it correctly, so please bear with me.

The old LDK node has stored a `u64` with the type `4`. So, if you now change the format, there will be a type mismatch between what LDK 124 has stored and what the future LDK is expecting.

If I’m understanding this correctly, you need a new type (a new number) for `manual_routing_params`. When upgrading, the `max_total_routing_fee_msat` will have a value, and from that, you can create the `manual_routing_params`. Then, you will store `manual_routing_params` as `Some` or `None`, depending on the value.

Reply:
Thanks a lot for the pointer, and guidance!
Reintroduced `max_total_routing_fee_msat` in [pr3342.03](https://github.com/shaavan/rust-lightning/commits/pr3342.03) to support downgrading.